tests/test_manager_api.py

```diff
@@ -111,6 +111,31 @@ def test_read_file(get_dapp):
             assert dapp.read_file(file_type) == f.read()
 
 
+@pytest.mark.parametrize("get_dapp", get_dapp_scenarios)
+def test_read_file_iter(get_dapp):
+    dapp = get_dapp(
+        [sys.executable, asset_path("write_mock_files.py")], state_file=True, data_file=True
+    )
+    sleep(0.5)
+
+    for file_type in ("state", "data"):
```

Comment:
It looks like two separated unit tests. Parameterizing can improve readability of this test